src/main/java/elise/Parser.java

```diff
@@ -0,0 +1,110 @@
+package elise;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ * Parser class to parse user inputs.
+ */
+public class Parser {
+    /**
+     * Parses user input into command.
+     *
+     * @param sc Scanner for system input.
+     * @return Command parsed.
+     */
+    protected static Command read(Scanner sc) throws EliseException {
+        int rank;
+        String command = sc.next();
+        String s = sc.nextLine().trim();
+        String[] message;
+
+        if (s.length() > 500) {
+            throw new EliseException("Message body is too long!");
+        }
+        switch (command) {
```

Comment:
Good that you practised the correct indentation for switch case, as it is a common mistake. Keep it up